conn.go

```diff
@@ -534,27 +544,7 @@ func (c *Conn) WriteControl(messageType int, data []byte, deadline time.Time) er
 		}
 	}
 
-	timer := time.NewTimer(d)
-	select {
-	case <-c.mu:
-		timer.Stop()
-	case <-timer.C:
-		return errWriteTimeout
```

Comment:
The new logic never returns `errWriteTimeout` if we can't get `c.mu` fast enough.  Does this not matter?

Reply:
Conn.write() is slightly less clever about that than WriteControl() was. write() could keep going longer than `deadline` would imply. Not sure how critical that is since the main write path we've been using for all our traffic never had it.